gpt.lua

```diff
@@ -0,0 +1,557 @@
+--@ module = true
+
+local json = require('json')
+local dfhack = require('dfhack')
+local utils = require('utils')
+local luasocket = require('plugins.luasocket')
+local gui = require('gui')
+local widgets = require('gui.widgets')
+local overlay = require('plugins.overlay')
+
+--
+-- TYPES
+--
+
+-- Enum for state of progress of the script.
+local Status = {
+  start = 0,
+  waiting = 1,
+  receiving = 2,
+  done = 3
+}
+
+local function string_from_Status(status)
+  if status == Status.start then return "start" end
+  if status == Status.waiting then return "waiting" end
+  if status == Status.receiving then return "receiving" end
+  if status == Status.done then return "done" end
+end
+
+local Content_Type = {
```

Comment:
this feels..odd, but I need to read through the code a few more times before I feel comfortable suggesting changes